libazureinit/src/user.rs

```diff
@@ -50,6 +50,29 @@ pub(crate) fn provision_ssh(
     Ok(())
 }
 
+pub fn add_user_for_passwordless_sudo(username: &str) -> Result<(), Error> {
+    // Create a file under /etc/sudoers.d with azure-init-user
+    let sudoers_path = "/etc/sudoers.d/azure-init-user";
+    let _ = File::create(sudoers_path)?;
+    let mut sudoers_file = std::fs::OpenOptions::new()
+        .append(true)
+        .create(true)
+        .open(sudoers_path)?;
+    write!(
+        sudoers_file,
+        "{} ALL=(ALL) NOPASSWD: ALL \n",
+        username.to_string()
+    )?;
+    sudoers_file.flush()?;
+    // Set the permission
+    let metadata = fs::metadata(sudoers_path)?;
+    let permissions = metadata.permissions();
+    let mut new_permissions = permissions.clone();
+    new_permissions.set_mode(0o700);
+    fs::set_permissions(sudoers_path, new_permissions)?;
```

Comment:
You can create a `Permissions` object from a mode constant, so you can do this instead:
```suggestion
    sudoers_file.set_permissions(Permissions::from_mode(0o600))?;
```
You'll also want to use `0o600` (owner read-write) instead of `0o700` (owner read-write-execute) since this isn't an executable.

Reply:
Done, added the suggestion